### PR TITLE
add sentry support

### PR DESCRIPTION
**Pull request by alexgul4enko**
sharing practice

### CODE REVIEW

File: src/app/common/hocs/withLists.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        if(results.length >= count || this.props.isLoading) {
+          return
+        }
+        const { page, offset, limit } = this.props.filters || {}
+        if(this.props.usePages) {
+          return this.props.fetch({ ...this.props.filters, page: page + 1, ...this.props.apiParams })
+        }
+        if(this.props.useLimit) {
+          return this.props.fetch({
+            ...this.props.filters, offset: offset + limit, limit: 25, ...this.props.apiParams,
```
Comment (yarilchenko) [inline src/app/common/hocs/withLists.jsx]: `limit` used very often. I see that we can rewrite it at any time, but maybe will be better to set this value somewhere as constant?
Let's imagine that you would want to set it one time with value 30 and use this value in all project.

File: src/app/common/hocs/withModal.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { Component } from 'react'
+
+export default function withModal(ChildComponent) {
+  return class MadalWrapper extends Component {
+    constructor(props) {
+      super(props)
+      this.state = {
```
Comment (yarilchenko) [inline src/app/common/hocs/withModal.jsx]: maybe better to use hooks in this place. useState + useEffect for a callback fn.

Comment (yarilchenko) [reply src/app/common/hocs/withModal.jsx]: because of lifecicle hooks will be redundant in this place

File: package.json [hunk 3]
```diff
@@ -15,7 +15,6 @@
     "Oleksii Onyshchenko (https://github.com/Asfalot)",
     "Tanya Sebastiyan <tanya.sebastiyan@gmail.com> (https://github.com/sebastiyan)",
     "Taras Dihtenko (https://github.com/h0m1c1de)",
-    "Vyacheslav Binetsky <vyacheslav.binetsky@gmail.com> (https://github.com/38Slava)",
```
Comment (h0m1c1de) [inline package.json]: lol

File: src/app/common/hocs/withLists.jsx (hunk 1 above)
Comment (h0m1c1de) [reply src/app/common/hocs/withLists.jsx]: agree, default limit can be in env

File: src/app/pages/auth/LoginFormContainer.jsx [hunk 4]
```diff
@@ -1,32 +1,27 @@
...
-class LoginFormContainer extends PureComponent {
-  componentWillReceiveProps({ submitting, submitSucceeded, navigate }) {
-    if(!submitting && submitSucceeded) {
-      navigate('dashboard')
-    }
+class LoginFormContainer extends Component {
+  shouldComponentUpdate(nextProps, nextState) {
+    return false
   }
+  onSunmit = (data) => this.props.session.create(data)
```
Comment (h0m1c1de) [inline src/app/pages/auth/LoginFormContainer.jsx]: onSubmit

File: src/app/common/widgets/Loading.jsx [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export default function Loading({ isLoading, children }) {
+  return isLoading ? <div className='loading_wrapper'>loading..</div> : children
```
Comment (h0m1c1de) [inline src/app/common/widgets/Loading.jsx]: loading_wrapper -> loading-wrapper

File: src/app/api.js [hunk 6]
```diff
@@ -36,15 +33,16 @@ function hasFile(obj) {
 }
 
 class API {
-  constructor(endpoint) {
+  constructor(store, endpoint) {
```
Comment (legendar) [inline src/app/api.js]: bad idea. ideally API should not depends on store

File: presets/proxy.js [hunk 7]
```diff
@@ -32,8 +31,7 @@ function configureProxy() {
 }
 
 function makeProxyContext(paths, targetUrl) {
-  const urlData = url.parse(targetUrl)
-
+  const urlData = new URL(targetUrl)
```
Comment (alexgul4enko) [inline presets/proxy.js]: this changes where made because of deprecated node url.parse method

File: package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "frontend-skeleton",
-  "version": "3.0.0-beta",
+  "version": "4.0.0-beta",
```
Comment (legendar) [inline package.json]: the version will be updated in separate PR. please remove this for now.

File: .env.default [hunk 9]
```diff
@@ -43,3 +43,9 @@ STORAGE_KEY=ds
 
 # what to store, set empty or null to store all state
 CACHE_STATE_KEYS=["resource.session"]
+
+#Sentry configs
+#SENTRY_ORG=fe_sceleton
+#SENTRY_PROJECT=react
+#SENTRY_AUTH_TOKEN=4e03c798977a4c25b2168abbb8ad5d632b777e9bde934db3a9f6574b6403ef48
+#SENTRY_DNS = https://8013d8c0a75143728ceb60ea451d3ac8@sentry.io/1532358
```
Comment (legendar) [inline .env.default]: DNS -> DSN

File: webpack.config.babel.js [hunk 10]
```diff
@@ -83,13 +85,26 @@ module.exports = createConfig([
...
   }),
 
+
+  env('production', [
+    addPlugins([
+      new SentryWebpackPlugin({
+        include: '.',
+        ignoreFile: '.sentrycliignore',
+        ignore: ['node_modules', 'webpack.config.babel.js', 'presets'],
+        configFile: 'sentry.properties',
```
Comment (legendar) [inline webpack.config.babel.js]: 1. it seems that you already have `.sentrycliignore` file, so you don't need `ignore` prop here
2. there no `sentry.properties` file in this PR

File: CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,5 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+## [4.0.0-beta]
+- add sentry
```
Comment (legendar) [inline CHANGELOG.md]: we have list of supported features in readme file (`Supports` section)
please add sentry there. 
we don't need separate changelog file. the changelog will be based on commits and will be attached for each release  (see https://github.com/django-stars/frontend-skeleton/releases)

File: .env.default [hunk 12]
```diff
@@ -43,3 +43,9 @@ STORAGE_KEY=ds
 
 # what to store, set empty or null to store all state
 CACHE_STATE_KEYS=["resource.session"]
+
+#Sentry configs
+#SENTRY_ORG=fe_sceleton
+#SENTRY_PROJECT=react
+#SENTRY_AUTH_TOKEN=4e03c798977a4c25b2168abbb8ad5d632b777e9bde934db3a9f6574b6403ef48
+#SENTRY_DSN = https://8013d8c0a75143728ceb60ea451d3ac8@sentry.io/1532358
```
Comment (legendar) [inline .env.default]: 1. we only need DSN and ENVIRONMENT options here

2. please always set default values (empty value is allowed):
```
SENTRY_DSN=
SENTRY_ENVIRONMENT=dev
```

File: presets/sentry.js [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+import SentryWebpackPlugin from '@sentry/webpack-plugin'
+
+
+export default function(config) {
+  return group([
+    env('production', [
+      addPlugins([
+        new SentryWebpackPlugin({
+          include: '.',
+          ignoreFile: '.sentrycliignore',
```
Comment (legendar) [inline presets/sentry.js]: can we include `src/app` instead of list all other files/folders in ignore ?

File: webpack.config.babel.js [hunk 14]
```diff
@@ -61,7 +62,7 @@ module.exports = createConfig([
   setEnv([
     // pass env values to compile environment
     'API_URL', 'AUTH_HEADER', 'MAIN_HOST',
-    'CACHE_STATE_KEYS', 'STORAGE_KEY',
+    'CACHE_STATE_KEYS', 'STORAGE_KEY', 'SENTRY_DSN',
```
Comment (legendar) [inline webpack.config.babel.js]: add `SENTRY_ENVIRONMENT` here

File: src/app/init.js [hunk 15]
```diff
@@ -6,12 +6,19 @@ import { routerReducer as router, routerMiddleware } from 'react-router-redux'
...
 import { middleware as cacheMiddleware, state as initialState } from './cache'
 import { reducers, epics } from 'store'
 import { reducer as resource, epic as resourceEpic } from 'common/utils/resource'
 import API, { configure as configureAPI } from 'api'
+import * as Sentry from '@sentry/browser'
+import createSentryMiddleware from 'redux-sentry-middleware'
 
 
+if(process.env.SENTRY_DSN) {
+  Sentry.init({ dsn: process.env.SENTRY_DSN })
```
Comment (legendar) [inline src/app/init.js]: add `environment` option here
